test/message-generator/mock/template-provider-mock-quick-template.json

```diff
@@ -0,0 +1,215 @@
+                "template_id": 30,
+                "future_template": true,
+                "version": 536870912,
+                "coinbase_tx_version": 2,
+                "coinbase_prefix": [3, 76, 163, 38, 0],
+                "coinbase_tx_input_sequence": 4294967295,
+                "coinbase_tx_value_remaining": 625000000,
+                "coinbase_tx_outputs_count": 0,
+                "coinbase_tx_outputs": [],
+                "coinbase_tx_locktime": 0,
+                "merkle_path": []
+            },
+            "id": "new_template2"
+        },
+        {
+            "message": {
+                "type": "SetNewPrevHash",
+                "template_id": 30,
+                "prev_hash": [145, 77, 225, 26, 186, 5, 16, 125, 174, 40, 238, 200, 210, 191, 188, 87, 191, 246, 242, 221, 8, 20, 202, 200, 97, 139, 241, 73, 137, 201, 28, 0],
+                "header_timestamp": 1671039088,
+                "n_bits": 545259519,
+                "target": [0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,255,255,127]
+            },
+            "id": "set_new_prev_hash2"
+        },
+        {
+            "message": {
+                "type": "RequestTransactionDataSuccess",
+                "template_id": 30,
+                "excess_data": [0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],
+                "transaction_list": []
+            },
+            "id": "request_transaction_data_success2"
+        }
+    ],
+    "frame_builders": [
+        {
+            "type": "automatic",
+            "message_id": "test/message-generator/messages/common_messages.json::setup_connection_success_template_distribution"
+        },
+        {
+            "type": "automatic",
+            "message_id": "new_template"
+        },
+        {
+            "type": "automatic",
+            "message_id": "set_new_prev_hash"
+        },
+        {
+            "type": "automatic",
+            "message_id": "new_template2"
+        },
+        {
+            "type": "automatic",
+            "message_id": "set_new_prev_hash2"
+        },
+        {
+            "type":"automatic",
+            "message_id": "request_transaction_data_success"
+        },
+        {
+            "type":"automatic",
+            "message_id": "request_transaction_data_success2"
+        }
+    ],
+    "actions": [
+        {
+            "message_ids": [],
+            "role": "server",
+            "results": [
+                {
+                    "type": "match_message_type",
+                    "value": "0x00"
+                }
+            ],
+            "actiondoc": "(pool) This action checks that a SetupConnection message is received"
+        },
+        {
+            "message_ids": ["setup_connection_success_template_distribution"],
+            "role": "server",
+            "results": [],
+            "actiondoc": "(pool) This action sends SetupConnection.Success with flag 2 (TD protocol)"
```

Comment:
the mock TP should expect 3 `SetupConnection` messages:
- pool
- mock JDS
- JDC

currently, it is only receiving one, and then moving onwards to the next messages

as a consequence, templates are sent downstream too fast, even before JDC has even started

Reply:
I am now spawning another TP at port 8443 to connect to JDC. Using the same file to connect to both the Pool and JDC doesn't seem to work. The overall flow looks good; only a few minor issues remain.